game/games/sortit/ui/player_selector.gd

```diff
@@ -7,9 +7,17 @@ var _player_inputs = []
 onready var _players = $MarginContainer/VBoxContainer/MarginContainer/Players
 onready var _back_button: Button = $MarginContainer/VBoxContainer/Buttons/HBoxContainer/BackButton
 onready var _play_button: Button = $MarginContainer/VBoxContainer/Buttons/HBoxContainer2/PlayButton
+# Path is to long, but cannot cleanly be split into multiple lines to avoid line length limit
+# gdlint: ignore=max-line-length
+onready var _help_button: Button = $MarginContainer/VBoxContainer/HBoxContainer/HBoxContainer/HelpButton
 
 
 func _ready():
+	# Open help screen, if game has not been played yet
+	var game_data = GameManager.get_game_data()
+	if not game_data.has("played"):
```

Comment:
Bad style: You're storing a boolean value, but then only check its existence. This should read `if not game_data.has("played") or not game_data.played:`.

Reply:
I actually deliberately decided against this. The reason: Adds more code, when it doesn't change a thing. Why would the value every be false ?

But yeah maybe I'll reconsider this.

Reply:
@MarioLiebisch Would an empty string be more fitting ?

Reply:
I think a boolean is the correct way to do this. Thinking about leaving it out due to performance is simply premature optimization with no significant gain. In addition to that, an empty string could be interpreted as an actual value. I'm not sure how Godot interprets that or if that interpretation ever changes.  Is the value not set? Or is it set, but to an empty string? It certainly has the potential to cause confusion later on, whether Godot changes or not.

As for a scenario on why even care about a boolean value: Later on one might consider an option switch to reset all tutorials in a game. Considering this, it might even be an interesting idea to add an extra wrapper for this, something like a `game_data.seen_tutorial("unique-identifier")` that will return `false` only once (the first time it's called).

Reply:
Ok I just tested your code and I have to say : I would have implemented it, if it were that simple. However, the implementation you suggested does not work. A second `if` or `elif` is needed.

This is **not** JavaScript the value is not undefined, if you try to get a value of a dictionary that doesn't exist.
In GDScript this will create a `Invalid get index 'played' (on base 'Dictonary')` which you can't catch.

```gdscript
if not game_data.has("played"):
    game_data["played"] = false
    GameManager.save_game_data()
elif not game_data.played:
    game_data["played"] = false
    GameManager.save_game_data()
```
or 

```gdscript
var has_been_played = true
if not game_data.has("played"):
	has_been_played = false
elif not game_data.played:
	has_been_played = false
if not has_been_played:
	game_data["played"] = false
	GameManager.save_game_data()
```

Doesn't look so nice, does it ? (Even if you create a `has_been_played` function, you would still add 5 extra lines of code in addition to the already existent if's)

Reply:
You're overcomplicating it. You can just write it this way:

```py
func _ready():
	# Open help screen, if game has not been played yet
	var game_data = GameManager.get_game_data()
	if not game_data.has("played") or not game_data.played:
		_on_help_button_up()

	# ...

func _on_play_button_up():
	# Mark game as played, to not show the help popup again by default
	var game_data = GameManager.get_game_data()
	if not game_data.has("played") or not game_data.played:

	# ...
```

GDScript – as well as many other languages – will evaluate an `or` from left to right. If the left side is `true`, it won't even bother trying to evaluate the right side, that's why you can just access `game_data.played` here, even if it actually doesn't exist (as in that case the left side would be true).

But as mentioned, just a style thing, so won't insist on it either. 🙂

Reply:
Ah okay, I actually thought that GDScript didn't do that. My bad. I actually tested `if not game_data.has("played") and game_data["played"] == false` since I think that makes it clear that a boolean is required here (Since Godot is dynamically typed (Sadly)). Obviously that would break the optimization that you just described here.

Reply:
I honestly don't know what I did wrong the first time, but I truly got a `Invalid get index 'played' (on base 'Dictonary')` exception. After testing it again, it now works.